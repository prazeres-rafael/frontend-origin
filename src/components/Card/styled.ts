import styled from 'styled-components';
import { Card, CardContent } from '@mui/material';

export const BaseCard = styled(Card)({
  boxShadow: '0px 16px 32px rgba(30, 42, 50, 0.08)',
  borderRadius: 8,
  width: '100%',

  '@media (min-width: 768px)': {
    width: '560px',
  },
});

export const BaseCardContent = styled(CardContent)({
  padding: '40px 24px',

  ':last-child': {
    paddingBottom: 40,
  },

  '@media (min-width: 768px)': {
    padding: '40px',
  },
});
