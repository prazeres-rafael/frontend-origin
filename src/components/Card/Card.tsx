import React from 'react';
import { BaseCard, BaseCardContent } from './styled';

type Props = {
  children?: React.ReactNode;
};

export default function Card({ children }: Props): React.ReactElement {
  return (
    <BaseCard>
      <BaseCardContent>{children}</BaseCardContent>
    </BaseCard>
  );
}
