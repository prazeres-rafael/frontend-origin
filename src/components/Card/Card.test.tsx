import { render } from '@testing-library/react';
import React from 'react';

import Card from './Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Card />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
