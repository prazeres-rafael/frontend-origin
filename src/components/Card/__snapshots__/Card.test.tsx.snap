// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
.c0 {
  box-shadow: 0px 16px 32px rgba(30,42,50,0.08);
  border-radius: 8px;
  width: 100%;
}

.c1 {
  padding: 40px 24px;
}

.c1:last-child {
  padding-bottom: 40px;
}

@media (min-width:768px) {
  .c0 {
    width: 560px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 40px;
  }
}

<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root c1 css-46bh2p-MuiCardContent-root"
  />
</div>
`;
